backend/manager/modules/vdsbroker/src/test/java/org/ovirt/engine/core/vdsbroker/monitoring/VirtMonitoringStrategyTest.java

```diff
@@ -172,6 +193,20 @@ public void testNeedToProcessHardwareCapsTrue() {
         newVds.setCpuFlags("flag2");
         assertTrue(virtStrategy.processHardwareCapabilitiesNeeded(oldVds, newVds));
     }
+    private  ImageTransferDao mockImageTransferDao() {
+        ImageTransferDao mock = mock(ImageTransferDao.class);
+        List<ImageTransfer> imageTransfers = new ArrayList<>();
+        ImageTransfer imageTransfer = new ImageTransfer();
+        imageTransfer.setPhase(ImageTransferPhase.FINISHED_SUCCESS);
+        imageTransfers.add(imageTransfer);
+        when(mock.getByVdsId(vdsId3)).thenReturn(null);
+        when(mock.getByVdsId(vdsId4)).thenReturn(imageTransfers);
+        imageTransfers.remove(0);
```

Comment:
better to create another list - I believe that when you modify the list here, it affects `when(mock.getByVdsId(vdsId4)).thenReturn(imageTransfers);`